File: example/code_java/src/main/java/app/Logger.java
package app;

public class Logger {


    public static void logCurrentTest(long received, long n) {
        System.err.println("> fibonacci(" + n + ") = " + received);
    }

    public static void logNoMatch(long expected) {
        System.err.println("----------------------------------------------------");
        System.err.println("The result does not match the expected value: " + expected);
        System.err.println("----------------------------------------------------");
    }

    public static void logException(Exception e) {
        System.err.println("----------------------------------------------------");
        System.err.println("An error occurred during runtime.");
        System.err.println("Details:");
        System.err.println(e.getMessage());
    }

    public static void logException(Throwable throwable) {
        System.err.println("----------------------------------------------------");
        System.err.println("Something bad happened!");
        System.err.println(throwable.getMessage());
        throwable.printStackTrace();
        System.err.println("----------------------------------------------------");
    }

    public static void logSuccess() {
        System.out.println("Everything went well, good job!");
        System.out.println("----------------------------------------------------");
    }
}

File: example/code_kotlin/src/main/kotlin/app/Logger.kt
package app

class Logger {
    companion object {
        fun logCurrentTest(received: Long, n: Long): Unit =
            System.err.println("> fibonacci($n) = $received")

        fun logNoMatch(expected: Long): Unit =
                System.err.println("""
                    ----------------------------------------------------
                    The result does not match the expected value: $expected
                    ----------------------------------------------------
                """.trimIndent())

        fun logException(exception: Exception): Unit =
                System.err.println("""
                    ----------------------------------------------------
                    An error occurred during runtime.
                    Details:
                    ${exception.message}
                """.trimIndent())

        fun logException(throwable: Throwable): Unit {
            System.err.println("""
                    ----------------------------------------------------
                    Something bad happened!
                    ${throwable.message}
                """.trimIndent())
            throwable.printStackTrace()
            System.err.println("----------------------------------------------------")
        }

        fun logSuccess(): Unit =
                println("""
                    Everything went well, good job!
                    ----------------------------------------------------
                """.trimIndent())
    }
}
